### ISSUE TITLE
[package] libbacktrace/cci.20210118: wrong autoconf version

### ISSUE THREAD

**Issue by oleurodecision** [conan-io/conan-center-index#5608]
### Package and Environment Details (include every applicable attribute)
  * Package Name/Version: **libbacktrace/cci.20210118**
  * Operating System+version: **Linux RedHat 6**
  * Compiler+version: **GCC 6**
  * Conan version: **conan 1.36.0**
  * Python version: **Python 3.7.8**

### Conan profile (output of `conan profile show default`)
```
Configuration for profile default:

[settings]
os=Linux
os_build=Linux
arch=x86
arch_build=x86
compiler=gcc
compiler.version=6
compiler.libcxx=libstdc++11
build_type=Release
[options]
[build_requires]
[env]
```

### Steps to reproduce (Include if Applicable)

  * create conanfile.txt with following content  
```
[requires]
libbacktrace/cci.20210118
```
  * call conan  
```
conan install --update --build outdated --settings build_type=Release --install-folder install .
```

It fails on wrong autoconf version, 2.71 is used, 2.69 seems to be mandatory for libbacktrace.

### Logs (Include/Attach if Applicable)
<details><summary>Click to expand log</summary>

```
$ conan install --update --build outdated --settings build_type=Release --install-folder install . 
Configuration:
[settings]
arch=x86
arch_build=x86
build_type=Release
compiler=gcc
compiler.libcxx=libstdc++11
compiler.version=6
os=Linux
os_build=Linux
[options]
[build_requires]
[env]

libbacktrace/cci.20210118: Not found in local cache, looking in remotes...
libbacktrace/cci.20210118: Trying with 'conan-center'...
Downloading conanmanifest.txt completed [0.28k]                                          
Downloading conanfile.py completed [3.76k]                                               
Downloading conan_export.tgz completed [0.36k]                                           
Decompressing conan_export.tgz completed [0.00k]                                         
libbacktrace/cci.20210118: Downloaded recipe revision 0
m4/1.4.18: WARN: Can't update, no package in remote
m4/1.4.18: Package is up to date
autoconf/2.71: Package is up to date
automake/1.16.3: Package is up to date
libtool/2.4.6: WARN: Can't update, no package in remote
libtool/2.4.6: Package is up to date
conanfile.txt: Installing package
Requirements
    libbacktrace/cci.20210118 from 'conan-center' - Downloaded
Packages
    libbacktrace/cci.20210118:920dfcba3d122dc23101db128649ffb81d98bdc4 - Build
Build requirements
    autoconf/2.71 from 'conan-center' - Cache
    automake/1.16.3 from 'conan-center' - Cache
    libtool/2.4.6 from 'conan-center' - Cache
    m4/1.4.18 from 'conan-center' - Cache
Build requirements packages
    autoconf/2.71:cb054d0b3e1ca595dc66bc2339d40f1f8f04ab31 - Cache
    automake/1.16.3:258f6f05ca54813b3d7180161753df7f9ccf7e1d - Cache
    libtool/2.4.6:6ebe207e83b263dcc5c1c76aa690035e8287e63a - Cache
    m4/1.4.18:6173abd988ee2e3159da6cce836868055fdb1cb4 - Cache

Installing (downloading, building) binaries...
m4/1.4.18: Already installed!
m4/1.4.18: Appending PATH environment variable: /home/jenkins/.conan/data/m4/1.4.18/_/_/package/6173abd988ee2e3159da6cce836868055fdb1cb4/bin
m4/1.4.18: Setting M4 environment variable: /home/jenkins/.conan/data/m4/1.4.18/_/_/package/6173abd988ee2e3159da6cce836868055fdb1cb4/bin/m4
autoconf/2.71: Already installed!
autoconf/2.71: Appending PATH env var with : /home/jenkins/.conan/data/autoconf/2.71/_/_/package/cb054d0b3e1ca595dc66bc2339d40f1f8f04ab31/bin
autoconf/2.71: Setting AC_MACRODIR to /home/jenkins/.conan/data/autoconf/2.71/_/_/package/cb054d0b3e1ca595dc66bc2339d40f1f8f04ab31/bin/share/autoconf
autoconf/2.71: Setting AUTOCONF to /home/jenkins/.conan/data/autoconf/2.71/_/_/package/cb054d0b3e1ca595dc66bc2339d40f1f8f04ab31/bin/autoconf
autoconf/2.71: Setting AUTORECONF to /home/jenkins/.conan/data/autoconf/2.71/_/_/package/cb054d0b3e1ca595dc66bc2339d40f1f8f04ab31/bin/autoreconf
autoconf/2.71: Setting AUTOHEADER to /home/jenkins/.conan/data/autoconf/2.71/_/_/package/cb054d0b3e1ca595dc66bc2339d40f1f8f04ab31/bin/autoheader
autoconf/2.71: Setting AUTOM4TE to /home/jenkins/.conan/data/autoconf/2.71/_/_/package/cb054d0b3e1ca595dc66bc2339d40f1f8f04ab31/bin/autom4te
autoconf/2.71: Setting AUTOM4TE_PERLLIBDIR to /home/jenkins/.conan/data/autoconf/2.71/_/_/package/cb054d0b3e1ca595dc66bc2339d40f1f8f04ab31/bin/share/autoconf
automake/1.16.3: Already installed!
automake/1.16.3: Appending PATH env var with : /home/jenkins/.conan/data/automake/1.16.3/_/_/package/258f6f05ca54813b3d7180161753df7f9ccf7e1d/bin
automake/1.16.3: Setting ACLOCAL to /home/jenkins/.conan/data/automake/1.16.3/_/_/package/258f6f05ca54813b3d7180161753df7f9ccf7e1d/bin/aclocal
automake/1.16.3: Setting AUTOMAKE_DATADIR to /home/jenkins/.conan/data/automake/1.16.3/_/_/package/258f6f05ca54813b3d7180161753df7f9ccf7e1d/bin/share
automake/1.16.3: Setting AUTOMAKE_LIBDIR to /home/jenkins/.conan/data/automake/1.16.3/_/_/package/258f6f05ca54813b3d7180161753df7f9ccf7e1d/bin/share/automake-1.16
automake/1.16.3: Setting AUTOMAKE_PERLLIBDIR to /home/jenkins/.conan/data/automake/1.16.3/_/_/package/258f6f05ca54813b3d7180161753df7f9ccf7e1d/bin/share/automake-1.16
automake/1.16.3: Setting AUTOMAKE to /home/jenkins/.conan/data/automake/1.16.3/_/_/package/258f6f05ca54813b3d7180161753df7f9ccf7e1d/bin/automake
automake/1.16.3: Append M4 include directories to AUTOMAKE_CONAN_INCLUDES environment variable
libtool/2.4.6: Already installed!
libtool/2.4.6: Appending PATH env: /home/jenkins/.conan/data/libtool/2.4.6/_/_/package/6ebe207e83b263dcc5c1c76aa690035e8287e63a/bin
libtool/2.4.6: Setting LIBTOOLIZE env to /home/jenkins/.conan/data/libtool/2.4.6/_/_/package/6ebe207e83b263dcc5c1c76aa690035e8287e63a/bin/libtoolize
libtool/2.4.6: Setting LIBTOOL_PREFIX environment variable to /home/jenkins/.conan/data/libtool/2.4.6/_/_/package/6ebe207e83b263dcc5c1c76aa690035e8287e63a
libtool/2.4.6: Setting LIBTOOL_DATADIR environment variable to /home/jenkins/.conan/data/libtool/2.4.6/_/_/package/6ebe207e83b263dcc5c1c76aa690035e8287e63a/bin/share
libtool/2.4.6: Setting LIBTOOL_PKGAUXDIR environment variable to /home/jenkins/.conan/data/libtool/2.4.6/_/_/package/6ebe207e83b263dcc5c1c76aa690035e8287e63a/bin/share/libtool/build-aux
libtool/2.4.6: Setting LIBTOOL_PKGLTDLDIR environment variable to /home/jenkins/.conan/data/libtool/2.4.6/_/_/package/6ebe207e83b263dcc5c1c76aa690035e8287e63a/bin/share/libtool
libtool/2.4.6: Setting LIBTOOL_ACLOCALDIR environment variable to /home/jenkins/.conan/data/libtool/2.4.6/_/_/package/6ebe207e83b263dcc5c1c76aa690035e8287e63a/bin/share/aclocal
libtool/2.4.6: Appending ACLOCAL_PATH env: /home/jenkins/.conan/data/libtool/2.4.6/_/_/package/6ebe207e83b263dcc5c1c76aa690035e8287e63a/bin/share/aclocal
libtool/2.4.6: Appending AUTOMAKE_CONAN_INCLUDES environment variable: /home/jenkins/.conan/data/libtool/2.4.6/_/_/package/6ebe207e83b263dcc5c1c76aa690035e8287e63a/bin/share/aclocal
libbacktrace/cci.20210118: Applying build-requirement: libtool/2.4.6
libbacktrace/cci.20210118: Applying build-requirement: automake/1.16.3
libbacktrace/cci.20210118: Applying build-requirement: autoconf/2.71
libbacktrace/cci.20210118: Applying build-requirement: m4/1.4.18
Downloading conan_sources.tgz completed [0.89k]                                          
Decompressing conan_sources.tgz completed [0.00k]                                        
libbacktrace/cci.20210118: Configuring sources in /home/jenkins/.conan/data/libbacktrace/cci.20210118/_/_/source
Downloading dedbe13fda00253fe5d4f2fb812c909729ed5937.tar.gz completed [649.57k]                        

libbacktrace/cci.20210118: Copying sources to build folder
libbacktrace/cci.20210118: Building your package in /home/jenkins/.conan/data/libbacktrace/cci.20210118/_/_/build/920dfcba3d122dc23101db128649ffb81d98bdc4
libbacktrace/cci.20210118: Generator txt created conanbuildinfo.txt
libbacktrace/cci.20210118: Calling build()
autoreconf: export WARNINGS=
autoreconf: Entering directory '.'
autoreconf: configure.ac: not using Gettext
autoreconf: running: /home/jenkins/.conan/data/automake/1.16.3/_/_/package/258f6f05ca54813b3d7180161753df7f9ccf7e1d/bin/aclocal --force -I config
configure.ac:33: error: Please use exactly Autoconf 2.69 instead of 2.71.
config/override.m4:12: _GCC_AUTOCONF_VERSION_CHECK is expanded from...
configure.ac:33: the top level
autom4te: error: /home/jenkins/.conan/data/m4/1.4.18/_/_/package/6173abd988ee2e3159da6cce836868055fdb1cb4/bin/m4 failed with exit status: 1
aclocal: error: /home/jenkins/.conan/data/autoconf/2.71/_/_/package/cb054d0b3e1ca595dc66bc2339d40f1f8f04ab31/bin/autom4te failed with exit status: 1
autoreconf: error: /home/jenkins/.conan/data/automake/1.16.3/_/_/package/258f6f05ca54813b3d7180161753df7f9ccf7e1d/bin/aclocal failed with exit status: 1
libbacktrace/cci.20210118: 
libbacktrace/cci.20210118: ERROR: Package '920dfcba3d122dc23101db128649ffb81d98bdc4' build failed
libbacktrace/cci.20210118: WARN: Build folder /home/jenkins/.conan/data/libbacktrace/cci.20210118/_/_/build/920dfcba3d122dc23101db128649ffb81d98bdc4
ERROR: libbacktrace/cci.20210118: Error in build() method, line 84
	self.run("{} -fiv".format(tools.get_env("AUTORECONF")), win_bash=tools.os_info.is_windows)
	ConanException: Error 1 while executing /home/jenkins/.conan/data/autoconf/2.71/_/_/package/cb054d0b3e1ca595dc66bc2339d40f1f8f04ab31/bin/autoreconf -fiv
```

</details>

**Comment by SpaceIm** [comment]
Maybe the call to autoreconf should be removed, I don't understand why it was added in the first place.

**Comment by SpaceIm** [comment]
Could you try https://github.com/conan-io/conan-center-index/pull/5609 please?

I don't know if it's the best solution.

**Comment by abhi1010** [comment]
Getting the same error when doing a custom boost compilation.



```
conan create --build=missing -tf None -pr ${WORKSPACE}/profile.txt . boost/1.76.0@
```

This requires the following packages:

```log
Packages
    boost/1.76.0:fcc1d7bed423fcdbcfbda52b86d36e9fdd3c76bf - Build
    bzip2/1.0.8:ba220f49d0a7a3c4ffcc9c53b1d9522afe8ff322 - Cache
    libbacktrace/cci.20210118:0ffb85008ec744757d3876a92928e8e3f1595e57 - Build
    libiconv/1.16:0ffb85008ec744757d3876a92928e8e3f1595e57 - Cache
    libunwind/1.5.0:e0c980fb3d5e04afc03de668d842c6d23fd4e3d4 - Cache
    xz_utils/5.2.5:0ffb85008ec744757d3876a92928e8e3f1595e57 - Cache
    zlib/1.2.11:761abc959eabcf3ab35822a6e37e38e5063dbb95 - Cache
```
Having the `libbacktrace/cci.20210118` in the list means it ends up with the same issue. 
Applying the patch from #5609 does not seem to help as well.

<details>
  <summary>Click to see logs</summary>

```log
libbacktrace/cci.20210118: Calling build()
autoreconf: export WARNINGS=
autoreconf: Entering directory '.'
autoreconf: configure.ac: not using Gettext
autoreconf: running: /home/centos/.conan/data/automake/1.16.3/_/_/package/178ad0abd52ecce13fccad76b91cbefc22a1e6ae/bin/aclocal --force -I config
configure.ac:33: error: Please use exactly Autoconf 2.69 instead of 2.71.
config/override.m4:12: _GCC_AUTOCONF_VERSION_CHECK is expanded from...
configure.ac:33: the top level
autom4te: error: /home/centos/.conan/data/m4/1.4.18/_/_/package/a746cdf55483adb95aa75f6c1fa1e1abeea4ab11/bin/m4 failed with exit status: 1
aclocal: error: /home/centos/.conan/data/autoconf/2.71/_/_/package/d942309c23a3fcc5e551b785c211ff20054cc63b/bin/autom4te failed with exit status: 1
autoreconf: error: /home/centos/.conan/data/automake/1.16.3/_/_/package/178ad0abd52ecce13fccad76b91cbefc22a1e6ae/bin/aclocal failed with exit status: 1
libbacktrace/cci.20210118:
libbacktrace/cci.20210118: ERROR: Package '0ffb85008ec744757d3876a92928e8e3f1595e57' build failed
libbacktrace/cci.20210118: WARN: Build folder /home/centos/.conan/data/libbacktrace/cci.20210118/_/_/build/0ffb85008ec744757d3876a92928e8e3f1595e57
ERROR: libbacktrace/cci.20210118: Error in build() method, line 84
        self.run("{} -fiv".format(tools.get_env("AUTORECONF")), win_bash=tools.os_info.is_windows)
        ConanException: Error 1 while executing /home/centos/.conan/data/autoconf/2.71/_/_/package/d942309c23a3fcc5e551b785c211ff20054cc63b/bin/autoreconf -fiv
```

</details>

**Comment by SpaceIm** [comment]
What is the error with #5609?

**Comment by alex-176** [comment]
@SpaceIm with your fix i managed to build boost (Linux, gcc 7.5 armv7hf)

**Comment by oleurodecision** [comment]
Thanks for the quick answer. Did try the #5609 patch on **this** specific node, seems to solve the issue :+1:

**Comment by abhi1010** [comment]
@SpaceIm Somehow I still see the same error:

```log
libbacktrace/cci.20210118: Calling build()
autoreconf: export WARNINGS=
autoreconf: Entering directory '.'
autoreconf: configure.ac: not using Gettext
autoreconf: running: /home/centos/.conan/data/automake/1.16.3/_/_/package/178ad0abd52ecce13fccad76b91cbefc22a1e6ae/bin/aclocal --force -I config
configure.ac:33: error: Please use exactly Autoconf 2.69 instead of 2.71.
config/override.m4:12: _GCC_AUTOCONF_VERSION_CHECK is expanded from...
configure.ac:33: the top level
autom4te: error: /home/centos/.conan/data/m4/1.4.18/_/_/package/a746cdf55483adb95aa75f6c1fa1e1abeea4ab11/bin/m4 failed with exit status: 1
aclocal: error: /home/centos/.conan/data/autoconf/2.71/_/_/package/d942309c23a3fcc5e551b785c211ff20054cc63b/bin/autom4te failed with exit status: 1
autoreconf: error: /home/centos/.conan/data/automake/1.16.3/_/_/package/178ad0abd52ecce13fccad76b91cbefc22a1e6ae/bin/aclocal failed with exit status: 1
libbacktrace/cci.20210118:
libbacktrace/cci.20210118: ERROR: Package '0ffb85008ec744757d3876a92928e8e3f1595e57' build failed
libbacktrace/cci.20210118: WARN: Build folder /home/centos/.conan/data/libbacktrace/cci.20210118/_/_/build/0ffb85008ec744757d3876a92928e8e3f1595e57
ERROR: libbacktrace/cci.20210118: Error in build() method, line 84
        self.run("{} -fiv".format(tools.get_env("AUTORECONF")), win_bash=tools.os_info.is_windows)
        ConanException: Error 1 while executing /home/centos/.conan/data/autoconf/2.71/_/_/package/d942309c23a3fcc5e551b785c211ff20054cc63b/bin/autoreconf -fiv

```

This is after removing all conan packages, starting clean.

**Comment by SpaceIm** [comment]
Obviously it can't work like this, the PR is not merged yet. Please try by editing locally libbacktrace recipe.

**Comment by madebr** [comment]
@oleurodecision

Just to rule out things, can you try removing and re-installing autoconf+automake+libtool+m4?
autoconf 2.71 received a fix in #5512

**Comment by SpaceIm** [comment]
The important message is `configure.ac:33: error: Please use exactly Autoconf 2.69 instead of 2.71.`

**Comment by abhi1010** [comment]
Looks like I left out some details so confusion entailed. To clarify, I used your changes. To be really sure, I checked out your branch as well.

```
(venv) [root@043bb4a81b15]~/conan-center/recipes/boost/all# git remote -v
origin  git@github.com:SpaceIm/conan-center-index.git (fetch)
origin  git@github.com:SpaceIm/conan-center-index.git (push)
(venv) [root@043bb4a81b15]~/conan-center/recipes/boost/all# git status
On branch libbacktrace-no-autoreconf
Your branch is up to date with 'origin/libbacktrace-no-autoreconf'.

nothing to commit, working tree clean
```

My case might be a bit special because I am trying to compile it against GCC 11, on centos 7. Also, to add, my profile also does not belong to default. 

```
$> egrep "boost" ~/$DIR/profile.txt
boost:extra_b2_flags="cxxflags=-flto" "linkflags=-flto"
boost:fPIC=False
boost:without_test=True
boost:shared=True
```

I also ran `conan remove -f "*"` before starting over, to rule out prior builds affecting the problem. This means everything was running from scratch. 

Thanks for the help. I appreciate it.

**Comment by madebr** [comment]
> The important message is `configure.ac:33: error: Please use exactly Autoconf 2.69 instead of 2.71.`

`configure.ac`:
https://github.com/ianlancetaylor/libbacktrace/blob/4f57c999716847e45505b3df170150876b545088/configure.ac#L32-L33

`AC_PREREQ`:
https://www.gnu.org/software/autoconf/manual/autoconf-2.69/html_node/Versioning.html

So there's something else going on.

**Comment by SpaceIm** [comment]
I don't know, but I can reproduce this issue on macOS, but not on Windows. It's weird that CI of CCI didn't fail.

**Comment by SpaceIm** [comment]
There is a file `override.m4` in `config` folder of libbacktrace archive:

```m4
dnl Test for the exact version when AC_INIT is expanded.
dnl This allows to update the tree in steps (for testing)
dnl by putting
dnl   m4_define([_GCC_AUTOCONF_VERSION], [X.Y])
dnl in configure.ac before AC_INIT,
dnl without rewriting this file.
dnl Or for updating the whole tree at once with the definition above.
AC_DEFUN([_GCC_AUTOCONF_VERSION_CHECK],
[m4_if(m4_defn([_GCC_AUTOCONF_VERSION]),
  m4_defn([m4_PACKAGE_VERSION]), [],
  [m4_fatal([Please use exactly Autoconf ]_GCC_AUTOCONF_VERSION[ instead of ]m4_defn([m4_PACKAGE_VERSION])[.])])
])
m4_define([AC_INIT], m4_defn([AC_INIT])[
_GCC_AUTOCONF_VERSION_CHECK
])
```

**Comment by antagonist112358** [comment]
Same issue ^^

**Comment by oleurodecision** [comment]
> @oleurodecision
> 
> Just to rule out things, can you try removing and re-installing autoconf+automake+libtool+m4?
> autoconf 2.71 received a fix in #5512

same behaviour (`configure.ac:33: error: Please use exactly Autoconf 2.69 instead of 2.71.`)

**Comment by madebr** [comment]
Thanks @oleurodecision for the check.
@SpaceIm  pointed me to the code in m4 that is causing the problem.
It's good to know that the problem was not with the cci autoconf package.

I've opened https://github.com/ianlancetaylor/libbacktrace/issues/74 to have some input from the upstream package.
